sdk/python/kfp/_client.py

```diff
@@ -498,3 +498,15 @@ def get_pipeline(self, pipeline_id):
       Exception if pipeline is not found.
     """
     return self._pipelines_api.get_pipeline(id=pipeline_id)
+
+  def delete_pipeline(self, pipeline_id):
```

Comment:
This method is already available as `Client().pipelines.delete_pipeline(id)`.
We'd like to avoid having multiple ways to do the same thing.
WDYT?

Reply:
My 5 cents:
There is already a pattern on selecting what api methods to expose, even when there's no custom logic:
```
list_experiments()
list_pipelines()
list_runs()
get_run()
upload_pipeline()
get_pipeline()
```
Taking the above into consideration, since a user may use `kfp delete`, they could also expect to be able to use `Client().delete_pipeline()` in some python script.

Finally, since there is no `.pipelines`, `.runs`, or `.experiments`, but it's `._pipelines_api`, `._run_api`, `_experiment_api`, and `_upload_api`, it would make sense not making the user access fields that are _private_. [Of course, they always can access them, but I guess they are private for some reason, and that could be that some responses require parsing.]

Reply:
>Finally, since there is no .pipelines, .runs, or .experiments

They exist.

Reply:
Could you provide a link to the code? Can't seem to see them.
If they are there indeed, I'm sorry.

Reply:
Those members are auto-generated from the low-level client API.
Try `dir(kfp.CLient())` or write `client.` in Jupyter and press "Tab".
The generation code is here: https://github.com/kubeflow/pipelines/blob/b2662f2fce7d641e19187c0a1a7b057ec975c1f6/sdk/python/kfp/_client.py#L40